### PR TITLE
Handle error on core level

### LINKED ISSUE #2148: Error Handling on Micro Frontend Level
Luigi should provide an error handler mechanism on core level which can be used to implement a central error handler within the core app. The error handler receives all unhandled errors of the micro frontends, which are using the Luigi Client.
The Luigi Client should wrap window.onerror handler and propagate the error to the host frame

### PR DESCRIPTION
**Pull request by JohannesDoberer**
Fixes #2148 

How to test:
```
onRuntimeErrorHandler: {
   errorFn: (error)=>{
      console.log('for this node you get an error ', error);
   }
}
```
Add this to a node and force an runtime error on this node.
In browser console you should see the console log

### CODE REVIEW

File: core/src/App.html [hunk 1]
```diff
@@ -1388,7 +1388,14 @@
           e.data.data.id,
           e.data.data.operation,
           e.data.data.params
-        );
+        )
+      }
+
+      if('luigi-runtim-error-handling'===e.data.msg){
```
Comment (stanleychh) [inline core/src/App.html:1394]:
```suggestion
      if('luigi-runtime-error-handling'===e.data.msg){
```
Typo?  Same as [luigi-client.js line20](https://github.com/SAP/luigi/pull/2167/files#diff-52086c5914c2bb232f321aef8650f2d284bc19d0052b6b222e704d47bece9755R20)?

File: client/src/luigi-client.js [hunk 2]
```diff
@@ -2,15 +2,33 @@ import { lifecycleManager } from './lifecycleManager';
...
+          const msg = {
+            msg:'luigi-runtim-error-handling',
+            errorObj:{
+              filename: error.filename,
+              message: error.message,
+              lineno: error.lineno,
+              colno: error.colno,
+              error: error.error
+            }
+          }
```
Comment (stanleychh) [inline client/src/luigi-client.js:28]:
```suggestion
        window.addEventListener('error', ({ filename, message, lineno, colno, error })=>{
          const msg = {
            msg:'luigi-runtime-error-handling',
            errorObj:{ filename, message, lineno, colno, error }
            }
          }
```
Please try this way and see it's still working properly.

Comment (stanleychh) [reply client/src/luigi-client.js:28]: Or this way is even cleaner. Please double-check if it works fine.
```suggestion
        window.addEventListener('error', ({ filename, message, lineno, colno, error }: errorObj)=>{
          const msg = {
            msg:'luigi-runtime-error-handling',
            errorObj
            }
          }
```

File: client/src/luigi-client.js [hunk 3]
```diff
@@ -2,15 +2,33 @@ import { lifecycleManager } from './lifecycleManager';
...
  */
 class LuigiClient {
   constructor() {
-    if (window !== window.top && window.document.head.getAttribute('disable-luigi-history-handling') != 'true') {
-      history.pushState = history.replaceState.bind(history);
+    if (window !== window.top){
+      if(window.document.head.getAttribute('disable-luigi-history-handling') != 'true') {
+        history.pushState = history.replaceState.bind(history);
+      }
+      if (window.document.head.getAttribute('disable-luigi-runtime-error-handling') != 'true') {
```
Comment (stanleychh) [inline client/src/luigi-client.js:17]:
```suggestion
      if(window.document.head.getAttribute('disable-luigi-history-handling') !== 'true') {
        history.pushState = history.replaceState.bind(history);
      }
      if (window.document.head.getAttribute('disable-luigi-runtime-error-handling') !== 'true') {
```

**Review by stanleychh — APPROVED** [review]

File: docs/navigation-parameters-reference.md [hunk 4]
```diff
@@ -594,6 +594,21 @@ Below is Luigi's web component example configuration which shows 3 web component
 ```
 - **since**: 1.7.0
 
+### onRuntimeErrorHandler
+<!-- add-attribute:class:warning -->
+> **NOTE**:  The **onRuntimeErrorHandler** only works if the micro frontend uses Luigi client and [disable-luigi-runtime-error-handling](advanced-options-luigi-client.md#disable-luigi-core-runtime-error-handling) is not disabled. 
+
```
Comment (alexandra-simeonova) [inline docs/navigation-parameters-reference.md:600]:
```suggestion
### onRuntimeErrorHandler

<!-- add-attribute:class:warning -->
> **NOTE**:  The **onRuntimeErrorHandler** only works if the micro frontend uses Luigi Client and [disable-luigi-runtime-error-handling](advanced-options-luigi-client.md#disable-luigi-core-runtime-error-handling) is not disabled. 

```

File: docs/navigation-parameters-reference.md [hunk 5]
```diff
@@ -594,6 +594,21 @@ Below is Luigi's web component example configuration which shows 3 web component
...
+- **type**: object
+- **description**: gives you the possibility to handle errors on core level. The error handler receives all unhandled errors of the micro frontends, which are using the Luigi Client.
```
Comment (alexandra-simeonova) [inline docs/navigation-parameters-reference.md:602]:
```suggestion
- **description**: allows you to handle errors on the Core level. The error handler receives all unhandled errors from the micro frontends which are using the Luigi Client.
```

File: docs/navigation-parameters-reference.md [hunk 6]
```diff
@@ -594,6 +594,21 @@ Below is Luigi's web component example configuration which shows 3 web component
...
+- **attributes**:
+  - **errorFn**:(function) gets an object `error` as parameter. The properties of this objects are the properties of the [error event](https://developer.mozilla.org/en-US/docs/Web/API/ErrorEvent)
```
Comment (alexandra-simeonova) [inline docs/navigation-parameters-reference.md:604]:
```suggestion
  - **errorFn**:(function) gets an object `error` as parameter. The properties of this object are the properties of the [error event](https://developer.mozilla.org/en-US/docs/Web/API/ErrorEvent).
```

File: docs/advanced-options-luigi-client.md [hunk 7]
```diff
@@ -26,3 +26,16 @@ By default, Luigi Client navigation does not manipulate the browser history. Lui
   <body></body>
 </html>
 ```
+
+## Disable Luigi Core runtime error handling
+
+By default, Luigi Client listen to runtime errors and send the [error event](https://developer.mozilla.org/en-US/docs/Web/API/ErrorEvent) properties to Luigi Core.
```
Comment (alexandra-simeonova) [inline docs/advanced-options-luigi-client.md:32]:
```suggestion
By default, Luigi Client listens to runtime errors and sends the [error event](https://developer.mozilla.org/en-US/docs/Web/API/ErrorEvent) properties to Luigi Core. 
```

File: docs/advanced-options-luigi-client.md [hunk 8]
```diff
@@ -26,3 +26,16 @@ By default, Luigi Client navigation does not manipulate the browser history. Lui
   <body></body>
 </html>
 ```
+
+## Disable Luigi Core runtime error handling
+
+By default, Luigi Client listen to runtime errors and send the [error event](https://developer.mozilla.org/en-US/docs/Web/API/ErrorEvent) properties to Luigi Core. 
+If a navigation node has configured a property called [onRuntimeErrorHandler](navigation-parameters-reference.md#node-parameters) you have the possibility to handle errors on core level.
```
Comment (alexandra-simeonova) [inline docs/advanced-options-luigi-client.md:33]:
```suggestion
If a navigation node has configured a property called [onRuntimeErrorHandler](navigation-parameters-reference.md#node-parameters), you have the possibility to handle errors on the Luigi Core level. For example: 
```

Comment (alexandra-simeonova) [inline docs/advanced-options-luigi-client.md:33]:
```suggestion
If a navigation node has configured a property called [onRuntimeErrorHandler](navigation-parameters-reference.md#node-parameters#onruntimeerrorhandler) you have the possibility to handle errors on core level.
```

**Pull request merged**